### PR TITLE
EPMRPP-86303 || add creatable multiple autocomplete

### PR DESCRIPTION
**Pull request by Vadim73i**

### CODE REVIEW

**Review by Bam6ycha — CHANGES_REQUESTED** [review]

File: app/src/components/fields/dynamicFieldsSection/fields/multipleAutocompleteField.jsx [hunk 1]
```diff
@@ -36,16 +37,26 @@ export const MultipleAutocompleteField = ({
 
   const getRequestParams = (term) => ({ method: 'PUT', data: { term } });
 
-  const parseValueToString = (user) => (user ? user.name : '');
+  const parseValueToString = (option) => {
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/components/fields/dynamicFieldsSection/fields/multipleAutocompleteField.jsx:40]: Looks like the expressions are not identical.
In first option the empty ('') value of option.name will lead to returning `option`.
In your case we will get empty string ''.
I'm not sure what is the right condition here as it require some time to debug and check all use cases. Anyway thanks for suggestion.

File: app/src/components/integrations/elements/bts/utils.js [hunk 2]
```diff
@@ -17,14 +17,17 @@
...
+  CREATABLE_MULTIPLE_AUTOCOMPLETE_TYPE,
 } from 'components/fields/dynamicFieldsSection/constants';
 
 export const getDefectFormFields = (fields, checkedFieldsIds, values) =>
   fields
     .filter((item) => item.required || checkedFieldsIds[item.id])
     .map((item) => {
       const isAutocomplete =
-        item.fieldType === AUTOCOMPLETE_TYPE || item.fieldType === MULTIPLE_AUTOCOMPLETE_TYPE;
+        item.fieldType === AUTOCOMPLETE_TYPE ||
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/components/integrations/elements/bts/utils.js:28]: Let's leave it as is for now. We also have an `array` field type, so I plan to reduce the number of similar fields during redesign.

File: app/src/components/fields/dynamicFieldsSection/fields/multipleAutocompleteField.jsx [hunk 3]
```diff
@@ -36,16 +37,26 @@ export const MultipleAutocompleteField = ({
...
+    if (option) {
+      return option.name ? option.name : option;
+    } else {
+      return '';
+    }
+  };
+
+  const customiseNewSelectedValue = (value) =>
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/components/fields/dynamicFieldsSection/fields/multipleAutocompleteField.jsx:48]: I'll update to 'z' as we have such term in several places in the app.

File: app/src/components/fields/dynamicFieldsSection/fields/multipleAutocompleteField.jsx [hunk 4]
```diff
@@ -36,16 +37,26 @@ export const MultipleAutocompleteField = ({
...
+    typeof value === 'string' ? { name: value } : value;
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/components/fields/dynamicFieldsSection/fields/multipleAutocompleteField.jsx:49]: image
This suggestion isn't suitable for some cases.

**Review by Bam6ycha — APPROVED** [review]

**Review by AmsterGet — APPROVED** [review]

**Pull request merged**